src/main/scala/leon/verification/VerificationPhase.scala

```diff
@@ -170,14 +170,24 @@ object VerificationPhase extends SimpleLeonPhase[Program,VerificationReport] {
 
       s.assertVC(vc)
 
-      val satResult = s.check
+      var supported = true
+      val satResult = try {
```

Comment:
You probably want to move the try up to before assertVC as that's probably where the unsupported will be thrown.